.changeset/kind-schools-retire.md

```diff
@@ -0,0 +1,5 @@
+---
+'eslint-plugin-primer-react': major
+---
+
+[Breaking] Adds `no-unnecessary-components` lint rule and enables it by default. This may raise new (typically autofixable) lint errors in existing codebases.
```

Comment:
Note that I've marked this as a major change because the new rule is enabled by default, which may raise new lint errors in existing code.